metriks/metriks.go

```diff
@@ -0,0 +1,140 @@
+package metriks
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/armon/go-metrics"
+	"github.com/armon/go-metrics/datadog"
+)
+
+const timerGranularity = time.Millisecond
+
+// Init will initialize the internal metrics system with a Datadog statsd sink
+func Init(serviceName string, conf Config) error {
+	return InitTags(serviceName, conf, nil)
+}
+
+// InitTags behaves like Init but allows appending extra tags
+func InitTags(serviceName string, conf Config, extraTags []string) error {
+	sink, err := datadog.NewDogStatsdSink(statsdAddr(conf), conf.Name)
+	if err != nil {
+		return err
+	}
+
+	var tags []string
+	for k, v := range conf.Tags {
+		tags = append(tags, fmt.Sprintf("%s:%s", k, v))
+	}
+
+	if extraTags != nil {
+		for _, t := range extraTags {
+			tags = append(tags, t)
+		}
+	}
+
+	sink.SetTags(tags)
+
+	return InitWithSink(serviceName, sink)
+}
+
+// InitWithSink initializes the internal metrics system with custom sink
+func InitWithSink(serviceName string, sink metrics.MetricSink) error {
+	c := metrics.DefaultConfig(serviceName)
```

Comment:
I'm not sure the default config is what we want. 

https://github.com/armon/go-metrics/blob/58588f401c2cc130a7308a52ca3bc6c0a76db04b/start.go#L52-L68

- I'm not sure if the `servicename` is going to end up as a prefix. I am not seeing it set `EnableServiceLabel`. 
- The `EnableHostname` says that it will prefix, which would cause a lot of trouble in the k8s world. I'd disable it.  
- The profiling stuff I wouldn't use, this one I'm more on the fence for this one.

Reply:
We are prefixing metric names by the `serviceName`, this is sort of nice because DDOG makes it difficult to search for metrics by tag in the dashboard UI. So you can type "eventeer" and find all eventeer metrics.

It's a good question on the hostname, I'll need to double check. We are definitely *not* prefixing metric names with the hostname and instead adding a `host` label. However, it does seem like that shouldn't be the default behavior if I'm reading this right so we probably want to be explicit. Either way, in our current configs we are adding the pod name as a label which is more useful than the hostname (which is already added by the ddog agent). So I'd vote to remove the host entirely by default here.

I find the profiling useful for general performance data. The number of goroutines can often grow large in unexpected scenarios and be a source of problems, I like to have it on a dashboard somewhere just in case.

Reply:
Regarding the hostname prefix, indeed we *should* not prefix as the host is a dimension per se.
In Edge-State we did the following: 

```
c := gometrics.DefaultConfig(svcName)
	c.TimerGranularity = timerGranularity
	c.HostName = hostname

	c.EnableServiceLabel = false // Otherwise it will add the service as label instead of prefix of metric name
	c.EnableHostname = false     // Otherwise it adds the hostname to all metrics
```

Reply:
I think hostname doesn't make sense by default. Same with pod name. In the k8s world I think this could cause a large growth in our cardinality (b/c pods are ephemeral). I think on the edge it could be really valuable to have the hostname (why is this node failing vs this one passing). 

One idea (totally not a requirement) is that this would be to implement a config mgmt using a options style (`func Config(serviceName string, opts ...Option)`).